Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridRows.cs

```diff
@@ -605,6 +605,13 @@ internal void RefreshRows(bool recycleRows, bool clearRows)
             }
         }
 
+        internal void RemoveRowAt(int rowIndex)
+        {
+            IEnumerable<DataGridRow> rows = GetAllRows();
```

Comment:
We don't need to grab all the rows here, we have access to `_rowsPresenter.Children` directly to get the specific index of the existing child.

Reply:
Committed changes